File: jbake-core/src/main/java/org/jbake/template/TemplateEngineAdapter.java
package org.jbake.template;

/**
 * Adapts model extractor output to used template engine.
 * This method typically wraps results of model extractions into data types suited to template engine
 *
 * @author ndx
 */
public interface TemplateEngineAdapter<Type> {

    class NoopAdapter implements TemplateEngineAdapter<Object> {

        @Override
        public Object adapt(String key, Object extractedValue) {
            return extractedValue;
        }

    }

    /**
     * Adapt value to expected output
     *
     * @param key            Template key
     * @param extractedValue Value to be used in template model
     * @return Value adapted for use in template
     */
    Type adapt(String key, Object extractedValue);

}

File: jbake-core/src/main/java/org/jbake/template/TemplateEngineAdapter.kt
package org.jbake.template

/**
 * Adapts model extractor output to used template engine.
 * This method typically wraps results of model extractions into data types suited to template engine
 */
interface TemplateEngineAdapter<Type> {

    /**
     * Adapt value to expected output.
     *
     * @return Value adapted for use in template
     */
    fun adapt(key: String, extractedValue: Any): Type


    class NoopAdapter : TemplateEngineAdapter<Any> {
        override fun adapt(key: String, extractedValue: Any): Any {
            return extractedValue
        }
    }
}
